src/string.cr

```diff
@@ -2738,12 +2738,11 @@ class String
   end
 
   # ditto
-  def rindex(search : Regex, offset = 0)
-    offset += size if offset < 0
```

Comment:
Why is this removed? You must support negative indexing from the end of the string.

Reply:
The substring (`self[0..offset]`) handles negative indexes. There is no need to bother until we change that for a more performant implementation.

Reply:
So why are the specs for negative indexes removed, and why do we have

```
return nil unless 0 <= offset <= size
```

which always returns nil for negative indexes?

Reply:
Argh, true that won't work.

Reply:
I think I'll just replace the implementation as suggested above. And add more specs...